Packages/Components/Sources/AssetListView.swift

```diff
@@ -30,6 +30,16 @@ public protocol AssetListViewable {
     var action: ((AssetListAction) -> Void)? { get set }
 }
 
+extension AssetListViewable {
+    var isAssetEnabled: Bool {
+        switch rightView {
+        case .balance,
+            .copy: return false
+        case .toggle(let value): return value
```

Comment:
can you simplify this file a bit? moving `AssetImage` to another file and maybe some other things around too